packages/jaeger-ui/src/components/TracePage/TraceTimelineViewer/VirtualizedTraceView.tsx

```diff
@@ -358,24 +358,39 @@ export class VirtualizedTraceViewImpl extends React.Component<VirtualizedTraceVi
     const isDetailExpanded = detailStates.has(spanID);
     const isMatchingFilter = findMatchesIDs ? findMatchesIDs.has(spanID) : false;
     const showErrorIcon = isErrorSpan(span) || (isCollapsed && spanContainsErredSpan(trace.spans, spanIndex));
-    const criticalPathSections = criticalPath?.filter(each => {
-      if (isCollapsed) {
-        const allChildSpanIds = [spanID, ...childSpanIds];
-        // This function called recursively to find all descendants of a span
-        const findAllDescendants = (currentChildSpanIds: string[]) => {
-          currentChildSpanIds.forEach(eachId => {
-            const currentChildSpan = trace.spans.find(a => a.spanID === eachId)!;
-            if (currentChildSpan.hasChildren) {
-              allChildSpanIds.push(...currentChildSpan.childSpanIds);
-              findAllDescendants(currentChildSpan.childSpanIds);
-            }
-          });
-        };
-        findAllDescendants(childSpanIds);
-        return allChildSpanIds.includes(each.spanId);
+
+    // Define an array to store the IDs of the span and its descendants (if the span is collapsed)
+    const allRequiredSpanIds = [spanID];
+
+    // If the span is collapsed, recursively find all of its descendants.
+    if (isCollapsed) {
```

Comment:
Q: if any of this logic required when the span is NOT collapsed? If not, please refactor to:
```
const criticalPathToRender = isCollapsed ? mergeChildrenCP() : criticalPath;
```